modules/formatter-jsonapi/lib/link.js

```diff
@@ -6,91 +6,96 @@ module.exports = function (model, opts) {
+    topLevelLinker(links);
+  } else {
 
-  // To-many link relations that were not explicitly included.
-  // Added to `links` as a string URL reference
-  toManyWithoutIncludes.reduce(function (result, relationName) {
-    var link = '/' + primaryType + '/' + model.id + '/' + relationName;
-    result[relationName] = link;
-    return result;
-  }, links);
+    // To-one link relations that were not explictly included. For
+    // example, a record in a database of employees might look like this:
+    // {
+    //   "id": "1",
+    //   "name": "tyler",
+    //   "position_id": "1"
+    // }
+    // The output of that record in json-api notation would be:
+    // {
+    //   "id": "1",
+    //   "name": "tyler",
+    //   "links": {
+    //     "self": "/employees/1/links/position",
+    //     "related": "/employees/1/position"
+    //   }
+    // }
+    linkWithoutIncludes.reduce(function (result, relationName) {
+      var id = model.id;
+      var link = {
+        self: '/' + primaryType + '/' + id + '/links/' + relationName,
+        related: '/' + primaryType + '/' + id + '/' + relationName
+      };
+      result[relationName] = link;
+      return result;
+    }, links);
 
-  // Link relations that were explictly included, adding the associated
-  // resources to the top level "included" object
-  linkWithIncludes.reduce(function (result, relationName) {
-    var related = relate(model, relationName);
-    var type = related.model ? related.model.typeName : related.constructor.typeName;
-    var link = {
-      type: type
-    };
+    // Link relations that were explictly included, adding the associated
+    // resources to the top level "included" object
+    linkWithIncludes.reduce(function (result, relationName) {
+      var id = model.id;
+      var related = relate(model, relationName);
+      var relatedType = related.model ? related.model.typeName : related.constructor.typeName;
+      var link = {
+        self: '/' + primaryType + '/' + id + '/links/' + relationName,
+        related: '/' + primaryType + '/' + id + '/' + relationName
+      };
 
-    if (related.models) {
-      // if the related is an array, we have a hasMany relation
-      link.id = related.reduce(function (result, model) {
-        var id = String(model.id);
-        // exclude nulls and duplicates, the point of a links
-        // entry is to provide linkage to related resources,
-        // not a full mapping of the underlying data
-        if (id && result.indexOf(id) === -1) {
-          result.push(id);
+      if (related.models) {
+        // if the related is an array, we have a hasMany relation
+        link.linkage = related.reduce(function (result, model) {
+          var id = String(model.id);
+          // exclude nulls and duplicates, the point of a links
+          // entry is to provide linkage to related resources,
+          // not a full mapping of the underlying data
+          if (id && result.indexOf(id) === -1) {
```

Comment:
This indexOf check is no longer functional because it doesn't take types into account.